### PR TITLE
Added randomized simulation parameters generation

### LINKED ISSUE #392: Fix Parameter change via proposal
### Scenario:
Change value for the `instantiateAccess` parameter from `AccessTypeOnlyAddress` with an address to a value without address like `AccessTypeNobody`. 

### Issue
It fails with `invalid parameter value: address not allowed for this type`. 

### Cause
The `address` part from the previous state was not cleared by the gov parameter change proposal handler. Probably because it is set to an empty value.

This can be replicated in a modified `TestUpdateParamsProposal` test.

### PR DESCRIPTION
**Pull request by bragaz**
After I integrated `wasmd` inside [desmos](https://github.com/desmos-labs/desmos) I found out that the simulation tests fails due to the fact that the `RandomizedParams` function of `module.go` returns `nil`.
```
--- FAIL: TestAppImportExport (128.47s)
panic: UnmarshalJSON cannot decode empty bytes [recovered]
	panic: UnmarshalJSON cannot decode empty bytes
```
In order to solve this I added the `simulation` folder with the params generation and the randomized genesis that will be called inside the tests. This could be useful for those projects that run sims.

Closes #392.

### CODE REVIEW

**Review by ethanfrey — COMMENTED** [review]
Looks fine to me, but I don't understand the simulation system.

I'd love @alpe to take a quick look before merging.

File: x/wasm/simulation/params.go
Comment (alpe) [inline x/wasm/simulation/params.go:17]: You need a full json object here
```suggestion
				jsonBz, _ := json.Marshal(&param)
				return string(jsonBz)
```

File: x/wasm/simulation/params.go
Comment (alpe) [inline x/wasm/simulation/params.go:23]:
```suggestion
				return fmt.Sprintf("%q", param.String())
```

File: x/wasm/simulation/params.go
Comment (alpe) [inline x/wasm/simulation/params.go:29]:
```suggestion
				return fmt.Sprintf(`"%d"`, param)
```

**Review by alpe — CHANGES_REQUESTED** [review]
Thank you very much for adding these simulation data generators!  🌻 
I have run them locally but ended up with failures. Please see my comments to fix the return types.

**Comment by ethanfrey** [comment]
Since this is not used by wasmd internally, I would ask you to submit working test cases for simulation to wasmd (so we could at least manually validate it is correct). We won't focus much more energy on this PR, but happy to merge if it gets working, and then include it in the next release.

We are making a 0.15.0 with the new cosmos-sdk v0.41.0.
Happy to make a 0.15.1 point release with sims when it is working

**Comment by bragaz** [comment]
Thanks @alpe! Can I ask which tests did you run to end up with failures?

**Comment by alpe** [comment]
> Thanks @alpe! Can I ask which tests did you run to end up with failures?

In case somebody is following this PR. We talked on discord to move this forward. https://discord.com/channels/737637324434833438/737640672680607764/804352247026417724

**Review by alpe — COMMENTED** [review]
Thanks again for the updates. 🤗  Only some minor comments on the code although I could not make it work on my machine. It fails with a panic and no helpful errors. 
```
ERROR OUTPUT

exit status 1
panic: halting simulations
goroutine 36 [running]:
main.worker(0x1, 0xc000074a20, 0xc00028e000)
	/Users/alex/go/pkg/mod/github.com/cosmos/tools/cmd/runsim@v1.0.0/main.go:216 +0x7a5
main.main.func2(0xc0002a2010, 0xc000074a20, 0xc00028e000, 0x1)
	/Users/alex/go/pkg/mod/github.com/cosmos/tools/cmd/runsim@v1.0.0/main.go:160 +0x6b
created by main.main
	/Users/alex/go/pkg/mod/github.com/cosmos/tools/cmd/runsim@v1.0.0/main.go:158 +0xc1a
make: *** [test-sim-import-export] Error 2
```
Does it work for you?

File: contrib/devtools/Makefile
Comment (bragaz) [reply contrib/devtools/Makefile:1]: There are 11 contributors to the Makefile, should I include them one by one?

**Comment by bragaz** [comment]
> Thanks again for the updates. 🤗 Only some minor comments on the code although I could not make it work on my machine. It fails with a panic and no helpful errors.
> 
> ```
> ERROR OUTPUT
> 
> exit status 1
> panic: halting simulations
> goroutine 36 [running]:
> main.worker(0x1, 0xc000074a20, 0xc00028e000)
> 	/Users/alex/go/pkg/mod/github.com/cosmos/tools/cmd/runsim@v1.0.0/main.go:216 +0x7a5
> main.main.func2(0xc0002a2010, 0xc000074a20, 0xc00028e000, 0x1)
> 	/Users/alex/go/pkg/mod/github.com/cosmos/tools/cmd/runsim@v1.0.0/main.go:160 +0x6b
> created by main.main
> 	/Users/alex/go/pkg/mod/github.com/cosmos/tools/cmd/runsim@v1.0.0/main.go:158 +0xc1a
> make: *** [test-sim-import-export] Error 2
> ```
> 
> Does it work for you?

Above those cryptic lines you should also see this line:

```
Seed 7: FAILED
To reproduce run: go test ./app -run TestAppImportExport -Enabled=true -NumBlocks=50 -Genesis= -Verbose=true -Commit=true -Seed=7 -Period=5 -ExportParamsPath /var/folders/xs/g0l3kxkx5msct63qbx9shlsr0000gn/T/sim-logs-833631974/sim_params-7.json -ExportStatePath /var/folders/xs/g0l3kxkx5msct63qbx9shlsr0000gn/T/sim-logs-833631974/sim_state-7.json -v -timeout 24h
ERROR OUTPUT 
```
If you're using GoLand (or IntelliJ) it's pretty easy to get a clearer error log by editing the configuration of the `TestSimAppImportExport` by doing these steps:
1) Run the test from within the `sim_test.go` file
2) Edit the configuration and paste this `-Enabled=true -NumBlocks=50 -Genesis= -Verbose=true -Commit=true -Seed=7 -Period=5 -ExportParamsPath /var/folders/xs/g0l3kxkx5msct63qbx9shlsr0000gn/T/sim-logs-833631974/sim_params-7.json -ExportStatePath /var/folders/xs/g0l3kxkx5msct63qbx9shlsr0000gn/T/sim-logs-833631974/sim_state-7.json`
inside `Program arguments` and `-v -timeout 24h` inside `Go tool arguments` like [this](https://pasteboard.co/JMvE2t9.png)
3) Re run the configured test

This will print a much meaningful error showing an error on import referring to the `capability` module.
And that was why I've removed that lines in the first place, but since it will make no difference because it will fail anyway for the same bug (since wasmd is using `capability`) I'll leave it there where it was before deleting it.

Since it's weird to have a failing test to make sure that params generation works, I've added an extra `TestFullAppSimulation` that exited with code 0.
For the `capability` bug I think I will discuss with my team since we will also run into that bug and open an issue into the Cosmos repo.

File: contrib/devtools/Makefile
Comment (alpe) [reply contrib/devtools/Makefile:1]: Just a note like: this file was copied from the cosmos-sdk repo <url>. Thank you to the original authors. 
A lot of code was copied or heavily inspired by the cosmos-sdk repo authors. Just adding a 🤗  here and there is nice.

Comment (bragaz) [reply contrib/devtools/Makefile:1]: Added!

**Review by alpe — APPROVED** [review]

**Comment by alpe** [comment]
@bragaz thank you very much for this big effort to integrate simulations. Well done 👏 
I will  👀  for the SDK bug fix and discuss how we can integrate them into our release process.

**Pull request merged**
